starport/pkg/cosmosgen/templates/vuex/store/index.ts.tpl

```diff
@@ -92,55 +103,70 @@ export default {
 				dispatch(subscription.action, subscription.payload)
 			})
 		},
-		{{ range .Module.HTTPQueries }}async {{ .FullName }}({ commit, rootGetters, getters }, { options: { subscribe = false , all = false}, params: {...key}, query=null }) {
+		{{ range .Module.HTTPQueries }}async {{ .FullName }}({ commit, rootGetters, getters }, { options: { subscribe, all} = { subscribe:false, all:false}, params: {...key}, query=null }) {
 			try {
-				
-				let value = query?(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}} query)).data:(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}}{{ if (gt $i 0)}}, {{ end}} key.{{$a}} {{end}})).data
+				const queryClient=await initQueryClient(rootGetters)
+				let value
+				{{ if (index .Rules 0).HasBody}}
```

Comment:
We need to iterate through `.Rules` as decribed here https://github.com/tendermint/starport/issues/901 instead of only using the first item.

Reply:
OH!...now I get what you meant...

Reply:
do you have an example of this? Trying to understand what this means for the store data....

Reply:
Can you please use `additional_bindings` for one of the RPC funcs of your choosing and run the `starport build` cmd and check the generated `rest.js` for it?

You'll see a function being generated with the same name but each has a different suffix (`index+1`).

https://github.com/googleapis/googleapis/blob/ca1372c6d7bcb199638ebfdb40d2b2660bab7b88/google/api/http.proto#L182-L199

Reply:
Your description about the be correct.

Queries do not necessarily need to accept the same params.

Reply:
Oh , I get that part, I'm just referring to how the data i ssupposed to be handled....

In the example you linked above, the Vuex store behaviour when making the RPC call getMessage, would be to add an entry keyed by "GetMessageRequest" to the "Message" state object

I'm asking whether this holds true regardless of http endpoint used, since multiple endpoints map to the same RPC call.

Basically, data-mapping is currently done on the gRPC-level definitions (regardless of HTTP bindings). A single RPC call with the same input data (GetMessageRequest) will update the same part of the store. Just want to make sure this is always the case

Reply:
The logic path might end up differently depending on the endpoint even if the underying RPC is the same. This totally depends on the server side implementation.

Reply:
> Your description about the be correct.
> 
> Queries do not necessarily need to accept the same params.

Sorry, didnt quite get that. What do you mean they don't necessarily accept the same params? isnt the Request message of an RPC call fixed?

Reply:
But we can use the same place in the Vuex store for different endpoints of the same RPC. It should be safe if the server side implementation is following the best practises. I think this is users' responsibility.

Reply:
> Sorry, didnt quite get that. What do you mean they don't necessarily accept the same params? isnt the Request message of an RPC call fixed?

RPC message is fixed but different endpoints configured for the same RPC might have different mappings to the underlying request message. Therefore, the generated query functions for the endpoints can have different params signature.

Reply:
> The logic path might end up differently depending on the endpoint even if the underying RPC is the same. This totally depends on the server side implementation.

so if you're NOT using the http endpoints but gRPC directly, which version of the logic are you getting ? :D

Reply:
You get to use RPC function itself if not using gRPC Gateway and using gRPC Web or gRPC.